FirebaseInstallations/Source/Library/FIRInstallations.m

```diff
@@ -117,12 +119,40 @@ - (instancetype)initWithAppOptions:(FIROptions *)appOptions
   return self;
 }
 
++ (void)validateAppOptions:(FIROptions *)appOptions appName:(NSString *)appName {
+  NSMutableArray *missingFields = [NSMutableArray array];
+  if (appName.length < 1) {
+    [missingFields addObject:@"`FirebaseApp.name`"];
+  }
+  if (appOptions.APIKey.length < 1) {
+    [missingFields addObject:@"`FirebaseOptions.APIKey`"];
+  }
+  if (appOptions.googleAppID.length < 1) {
+    [missingFields addObject:@"`FirebaseOptions.googleAppID`"];
+  }
+  if (appOptions.GCMSenderID.length < 1) {
+    [missingFields addObject:@"`FirebaseOptions.GCMSenderID`"];
+  }
+
+  if (missingFields.count > 0) {
+    [NSException
+         raise:kFirebaseInstallationsErrorDomain
+        format:@"%@[%@] Could not configure Firebase Installations due to invalid FirebaseApp "
+               @"options. The following parameters are nil or empty: %@. If you use "
+               @"GoogleServices-Info.plist please download the most recent version from Firebase "
+               @"Console. If you configure Firebase in code, please make sure you specify all "
+               @"required parameters.",
```

Comment:
I recommend that we list the required parameters:
"APIKey and projectID"

Reply:
The list of the missing parameters will be added instead of `%@` at the string `"...The following parameters are nil or empty: %@..."` from `missingFields`